### ISSUE TITLE
This package is rendered superfluous by Node v10.12.0

### ISSUE THREAD

**Issue by TheGrandmother** [substack/node-mkdirp#164]
In Node 10.12 and above there is a `recursive` option available to `fs.mkdir` which makes `fs.mkdir` behave as `mkdir -p`.

**Comment by geotrev** [comment]
Does that include CLI as well? or is `mkdir` in a node env always going to supersede the linux/system command? asking for a friend.

**Comment by TheGrandmother** [comment]
@geotrev 
`mkdir` is present on every POSIX system and has the  `-p` flag so the command line version of this command is a bit superfluous from the beginning.
The posix `mkdir` command also supports setting accessmodes.

I'm willing to bet that the node implementation uses the posix [mkdir()](https://pubs.opengroup.org/onlinepubs/009695399/functions/mkdir.html) function.

**Comment by isaacs** [comment]
There are still good reasons to use `mkdirp`, especially if you need to know the first dir made, or care about getting error messages reliably.  Detailed in the readme here: https://github.com/isaacs/node-mkdirp#choosing-a-recursive-mkdir-implementation

@TheGrandmother I'd ask how much you're willing to bet, but it feels wrong to take money from a stranger ;P  I think you're confusing the `mkdir` _command_ present on posix command lines with the `mkdir` _function_ in libc.  (Ie, [`mkdir(1)`](https://linux.die.net/man/1/mkdir) vs [`mkdir(2)`](https://linux.die.net/man/2/mkdir).)  There is no built-in way to create dirs recursively from C.

**Comment by lydell** [comment]
Node v12+ gives you the first dir made:

>  v13.11.0, v12.17.0 | In recursive mode, the callback now receives the first created path as an argument.

https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback

**Comment by isaacs** [comment]
@lydell Yes, and I believe at some point it also fixed the broken error messages.

Nevertheless, node versions less than 12.17.0 exist, and we'll keep caring about them for a while.  npm 7 supports node 10.0.0.  It'll be quite some time before this module is superfluous, most likely a few years at least.  When it is in an environment where `fs.mkdir` can do everything this module needs to do, it just calls the node built-in `fs.mkdir`.
